Airport2/Airport2/Panel.cs

```diff
@@ -0,0 +1,856 @@
+                ref ticket
+            );
+
+            if (!succeed)
+            {
+                Panel.showError("Operation failed.");
+                return;
+            }
+
+            var passenger = new Passenger(firstName, secondName, nationality, passport, birthday, gender);
+            passenger.BuyTicket(flight, ticket);
+            flight.AddPassenger(passenger);
+        }
+        public void AskForDeletingPassenger()
+        {
+            var flightNumber = Guid.Empty;
+
+            Console.Write("Please, enter flight number: ");
+            var flightNumberToSearch = Console.ReadLine();
+            var succeed = Guid.TryParse(flightNumberToSearch, out flightNumber);
+            if (!succeed)
+            {
+                Panel.showError("Wrong flight number entered.");
+                Panel.showError("Operation failed.");
+                return;
+            }
+
+            var flight = this.SearchByFlightNumber(flightNumber);
+            if (flight == null)
+            {
+                Panel.showError($"There is no flight with number: {flightNumberToSearch}");
+                Panel.showError("Operation failed.");
+                return;
+            }
+
+            Console.Write("Please, enter passenger passport: ");
+            var passportToSearch = Console.ReadLine();
+            var passenger = Panel.SearchByPassengerPassport(flight, passportToSearch);
+            if (passenger == null)
+            {
+                Panel.showError("Wrong passenger passport entered.");
+                Panel.showError("Operation failed.");
+                return;
+            }
+
+            flight.RemovePassenger(passenger);
+        }
+
+        public Flight SearchByFlightNumber(Guid flightNumber)
+        {
+            foreach (var flight in this.Flights)
+                if (flight.Number == flightNumber)
+                    return flight;
+
+            return null;
+        }
+        public Flight[] SearchByFlightPrice(uint flightPrice)
+        {
+            var flights = new Flight[0];
+            var found = false;
+
+            foreach (var flight in this.Flights)
+                if (flight.Price == flightPrice)
+                {
+                    flights = Helper.ResizeArray(flights, flights.Length + 1);
+
+                    flights[flights.Length - 1] = flight;
+                    found = true;
+                }
+
+            return found ? flights : null;
+        }
+        public Flight[] SearchByFlightArrivalCity(City city)
+        {
+            var flights = new Flight[0];
+            var found = false;
+
+            foreach (var flight in this.Flights)
+                if (flight.ArrivalCity == city)
+                {
+                    flights = Helper.ResizeArray(flights, flights.Length + 1);
+                    flights[flights.Length - 1] = flight;
+                    found = true;
+                }
+
+            return found ? flights : null;
+        }
```

Comment:
SearchByFlightArrivalCity and SearchByFlightDepartureCity do the same. Why you didn`t make one method and reuse it?